src/event/mod.rs

```diff
@@ -267,6 +291,8 @@ impl Value {
             Value::Integer(num) => Bytes::from(format!("{}", num)),
             Value::Float(num) => Bytes::from(format!("{}", num)),
             Value::Boolean(b) => Bytes::from(format!("{}", b)),
+            Value::Map(_) => todo!(),
+            Value::Array(_) => todo!(),
```

Comment:
I'm curious how we use `as_bytes()` currently. It's clearly not a great fit for nested data, so ideally we'd be able to remove it from the API.

Reply:
So it turns out that it is used in many places over the code, such as config testing or computations of hashes, but most of the uses don't strictly require use of `as_bytes`. I think eventually we can just get rid of `as_bytes` at all (as a separate issue/PR), but for now it seems like using `serde_json::to_vec` to a sensible representation of fields which can be used for example for comparisons is fine.